packages/components/src/components/select-native/select-native.jsx

```diff
@@ -15,9 +15,25 @@ const getDisplayText = (list_items, value) => {
     return '';
 };
 
-const SelectNativeOptions = ({ list_items, should_hide_disabled_options, use_text }) => {
+const SelectNativeOptions = ({ label, list_items, should_hide_disabled_options, use_text }) => {
     const options = should_hide_disabled_options ? list_items.filter(opt => !opt.disabled) : list_items;
-
+    const grouped = options.reduce((r, a) => {
+        r[a.group] = [...(r[a.group] || []), a];
```

Comment:
Creating a new array every time is inefficient. We can push it to the exiting array instead.

Reply:
Understand, made suggested changes.